### PR TITLE
Added controller, service and repository methods and tests for PATCH /chats/:id

### PR DESCRIPTION
**Pull request by bvbps**

### CODE REVIEW

File: app/controllers/ChatController.scala [hunk 1]
```diff
@@ -57,6 +57,14 @@ class ChatController @Inject() (cc: ControllerComponents, chatService: ChatServi
     }
   }
 
+  def moveChatToTrash(chatId: String): Action[AnyContent] = authenticatedUserAction.async {
+    authenticatedRequest =>
+      chatService.moveChatToTrash(chatId, authenticatedRequest.userId).map {
+        case true => NoContent
```
Comment (IvoSaraiva) [inline app/controllers/ChatController.scala]: Don't pattern match on booleans. Use map(if (_) NoContent else NotFound)

File: test/repositories/slick/implementations/ChatsRepositorySpec.scala [hunk 2]
```diff
@@ -526,4 +526,33 @@ class ChatsRepositorySpec extends AsyncWordSpec with MustMatchers with BeforeAnd
...
+    val chatsRep = new SlickChatsRepository(db)
+
+    val userId = "148a3b1b-8326-466d-8c27-1bd09b8378f3" //beatriz@mail.com
+
+    "remove the user's chat from inbox, sent and draft and move it to trash" in {
+      val validChatId = "303c2b72-304e-4bac-84d7-385acb64a616"
+      for {
+        result <- chatsRep.moveChatToTrash(validChatId, userId)
+        optionUserChat <- db.run(UserChatsTable.all.filter(uc => uc.chatId === validChatId && uc.userId === userId).result.headOption)
+        userChat = optionUserChat.get
```
Comment (IvoSaraiva) [inline test/repositories/slick/implementations/ChatsRepositorySpec.scala]: Don't use get. Use the "Inside" trait from scala test.

File: test/repositories/slick/implementations/ChatsRepositorySpec.scala [hunk 3]
```diff
@@ -526,4 +526,33 @@ class ChatsRepositorySpec extends AsyncWordSpec with MustMatchers with BeforeAnd
...
+      } yield assert(
+        result &&
+          userChat.inbox === 0 &&
```
Comment (IvoSaraiva) [inline test/repositories/slick/implementations/ChatsRepositorySpec.scala]: What is the "===" method?

File: app/controllers/ChatController.scala (hunk 1 above)
Comment (joaov2) [reply app/controllers/ChatController.scala]: Is the use of ".map(if(_)...) " better just from a visual perspective, or is there a performance difference?

File: test/repositories/slick/implementations/ChatsRepositorySpec.scala (hunk 2 above)
Comment (bvbps) [reply test/repositories/slick/implementations/ChatsRepositorySpec.scala]: I didn't know that trait existed. I'm going to learn it right away :) Thanks!

Comment (joaov2) [reply test/repositories/slick/implementations/ChatsRepositorySpec.scala]: Wouldn't the OptionValues trait be more appropriate?

File: test/repositories/slick/implementations/ChatsRepositorySpec.scala (hunk 3 above)
Comment (bvbps) [reply test/repositories/slick/implementations/ChatsRepositorySpec.scala]: Found a good explaination:
"ScalaTest lets you use Scala's assertion syntax, but defines a triple equals operator (===) to give you better error messages. The following code would give you an error indicating only that an assertion failed:
assert(1 == 2) 
Using triple equals instead would give you the more informative error message, "1 did not equal 2":
assert(1 === 2)"

**Pull request merged**